FEC-9928 Manifest downloading issue in case of cross protocol redirection

GouravSna opened the pull request.

GouravSna committed:
- Handle permanent redirects

GouravSna committed:
- Added okhttp3 lib in dtglib
- Changed targetSdkVersion to 29
- manifest download code migrated from HttpUrlConnection to OkHttp
- Using crossProtocol redirection

GouravSna committed:
- Code refactor

GouravSna committed:
- Added configurable params for timeouts and cross protocol redirections to okhttp

noamtamim reviewed (changes requested):
I'd rather not add a dependency in okhttp at this point. If there's a real (production) need to support this feature, we can do what ExoPlayer does (read the redirect header and resend the request).

GouravSna committed:
- Removed okhttp
- Using httpurlconnection: cross protocol (ssl to nonssl and viceversa only) redirections are being handled manually

GouravSna committed:
- Removed unused variables from Settings
